src/polygon_filter.cpp

```diff
@@ -274,6 +274,8 @@ bool LaserScanPolygonFilterBase::configure()
   param_config.invert = invert_filter_;
   dyn_server_->updateConfig(param_config);
 
+  padPolygon(polygon_, polygon_padding);
```

Comment:
Instead of calling `padPolygon() ` here, could you move `setCallback(f)` from line 261 to here? By doing that, `reconfigureCB()` is called after the `polygon` parameter has manually been converted from an array to a string. Also could you add a comment saying something to the effect of "Calling setCallback(f) here calls `reconfigureCB()` which updates the polygon padding.

Reply:
I tested your suggestion and it works, so I pushed a commit to do just that. I agree, that this is a cleaner and more reasonable way to resolve the issue.